robocop/checkers/comments.py

```diff
@@ -50,6 +50,93 @@ def visit_KeywordCall(self, node):  # noqa
         self.find_comments(node)
         self.generic_visit(node)
 
+    def visit_Return(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Documentation(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_SectionHeader(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Tags(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Setup(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Timeout(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Template(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Arguments(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Variable(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_Metadata(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_ForceTags(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_DefaultTags(self, node):  # noqa
+        self.find_comments(node)
+        self.generic_visit(node)
+
+    def visit_IfHeader(self, node):  # noqa
```

Comment:
You don't need to check for IS_RF4 if whole node was introduced in 4.0+ - visitor will just not match this node if it doesn't exist

Reply:
Yeah, I thought the same but I was trying to understand why it visits some of the nodes even if they haven't existed in RF 3.2. For example lines: test.robot:38:16, test.robot:40:29, test.robot:42:10.

Reply:
It will visit those nodes but it will not recognize it as for loop - just as keyword call (with weird indentation ;) )